src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -288,43 +310,134 @@ private async Task SaveAsync()
                 this.pendingTipHashHeight = null;
             }
 
-            this.CheckItemsAreInSequence(pendingBatch.Keys.ToList());
+            this.CheckItemsAreInConsecutiveSequence(pendingBatch.Keys.ToList());
 
+            // Save the items to disk.
             using (new StopwatchDisposable(o => this.performanceCounter.AddInsertTime(o)))
             {
                 // Save the items to disk.
-                await this.provenBlockHeaderRepository.PutAsync(pendingBatch.Values.ToList(), hashHeight);
+                await this.provenBlockHeaderRepository.PutAsync(pendingBatch.Values.ToList(), hashHeight).ConfigureAwait(false);
+
+                this.TipHashHeight = this.provenBlockHeaderRepository.TipHashHeight;
+            }
+        }
+
+        /// <summary>
+        /// Will recover the <see cref="IProvenBlockHeaderStore"/> tip to the most recent <see cref="ProvenBlockHeader"/> 
+        /// that exists in the <see cref="IProvenBlockHeaderRepository"/> and/or <see cref="ChainedHeader"/>.
+        /// </summary>
+        /// <param name="newChainedHeader">Current <see cref="ChainedHeader"/> tip with all its ancestors.</param>
```

Comment:
this line doesn't mean anything

Reply:
Made it more clear - `New <see cref="ChainedHeader"/> to try and recover to.`